ast/ksp/src/main/kotlin/me/tatarka/kotlin/ast/KSAst.kt

```diff
@@ -110,7 +110,23 @@ private interface KSAstAnnotated : AstAnnotated {
     override fun annotationAnnotatedWith(packageName: String, simpleName: String): AstAnnotation? {
         return declaration.annotationAnnotatedWith(packageName, simpleName)?.let {
             KSAstAnnotation(resolver, it)
-        }
+        } ?: overrideeAnnotationAnnotatedWith(declaration, packageName, simpleName)
```

Comment:
hm, I wonder if doing this here would be confusing, especially since we are explicitly looking for overrides in other places in the code. I was thinking of doing it in [TypeCollector](https://github.com/evant/kotlin-inject/blob/81280fb80e6fd2040127e649d4afa267ad76e1bc/kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt) which is where we are collecting provides methods. It could check for scopes as well when checking for `@Provides` on the overriden method in [collectTypeInfo()](https://github.com/evant/kotlin-inject/blob/81280fb80e6fd2040127e649d4afa267ad76e1bc/kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt#L282) and use that information for obtaining the scope in [collectTypes()](https://github.com/evant/kotlin-inject/blob/81280fb80e6fd2040127e649d4afa267ad76e1bc/kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt#L103)

Reply:
I was going to ask about that (but got distracted making it work for properties). Unless there's certain cases where we don't want to look at super types, maybe it makes sense to just do it here?

Reply:
Hm, wouldn't be against it. But would want to change how `collectTypeInfo` works then so we aren't double-checking. And write a few more tests to make sure we didn't break anything.

Reply:
Er, also want to note that we need to scan superclass methods anyway regardless on if they override or not. The use of a hashmap for checking overrides was to prevent causing an n^2 lookup, though haven't check if `findOverridee()` does something smarter than linearly scanning methods for a match.

Reply:
I'm looking into it now, but the way that `@Provides` is currently handled might be the better approach. Also the docs for `findOverridee()` mention that:

> Calling `findOverridee` is expensive and should be avoided if possible

Reply:
Does it make sense for a `@Provides` or a class to have multiple scope annotations? If not, I think a more up front error could be clearer here.

The current behavior is that the first scope annotation is used, which leads to failures later if the first scope doesn't match the scope of the component that it is being used with.

Reply:
we error if there's conflicting scopes on the class, I think we should error in the same way here

Reply:
I think an error would be easier to understand there, we could always relax later if there's a real need for that pattern.